frontend2/src/components/team/EligibilitySettings.tsx

```diff
@@ -100,18 +113,44 @@ const EligibilitySettings: React.FC = () => {
                     desiredEligibility.includes(criterion.id),
                     (team.profile?.eligible_for ?? []).includes(criterion.id),
                   )}
-                  onChange={(checked) => {
+                  onChange={async (checked) => {
                     if (checked && !desiredEligibility.includes(criterion.id)) {
-                      setDesiredEligibility([
-                        ...desiredEligibility,
-                        criterion.id,
-                      ]);
+                      // Add criterion to team
+                      // We do NOT want to await this, because we want to send the state of this mutation to isPending!
+                      updateTeam({
+                        episodeId,
+                        patchedTeamPrivateRequest: {
+                          profile: {
+                            ...team.profile,
+                            eligible_for: [
+                              ...(team.profile?.eligible_for ?? []),
+                              criterion.id,
+                            ],
+                          },
+                        },
```

Comment:
i think there will be a race condition here - an user can click multiple eligibilities in quick succession, but team.profile.eligible_for doersn't update until a request returns - thus the eligibilities will all override each other. 
I prevented this before by using desiredEligibility as a way to track the overall desired state on the client side